test/rules/avoid_relative_lib_imports.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AvoidRelativeLibImportsTest);
+  });
+}
+
+@reflectiveTest
+class AvoidRelativeLibImportsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'avoid_relative_lib_imports';
+
+  test_packageImport() async {
+    newFile('$testPackageRootPath/lib/other.dart');
+    await assertNoDiagnostics(
+      r'''
+// ignore: unused_import
+import 'package:test/other.dart';
+''',
+    );
+  }
+
+  test_absolutePath() async {
+    var filePath = '$testPackageRootPath/lib/other.dart';
+    newFile(filePath);
+    await assertDiagnostics(
+      '''
+// ignore: unused_import
+import '$filePath';
+''',
+      [
+        lint('avoid_relative_lib_imports', 32, filePath.length + 2),
+        // TODO(dantup): Why???
```

Comment:
I'm not sure, but I will point out that the file path is an absolute path, so it's possible that we aren't handling that case correctly. @scheglov would probably know.

Reply:
I haven't figured out where to pause to track down what's happening here properly, but this looks very strange. The Import string is an absolute path, but the source it's looking for looks like it's been interpreted relative to the package root or something:

<img width="1330" alt="Screenshot 2022-02-10 at 18 32 27" src="https://user-images.githubusercontent.com/1078012/153473700-eefca746-c45f-4d4d-b8bc-a6d70ab6967e.png">

That makes me wonder whether this test is valid - do we expected to support absolute paths like this? (I thought we did, but upon reflection, it seems like a very odd thing for someone to write).